### PR TITLE
Rename context filters to scope filters

### PR DESCRIPTION
**Pull request by yanivagman**
### 1. Explain what the PR does

This commit updates the terminology within the Tracee codebase, replacing occurrences of "context filters" with "scope filters". By doing so, it enhances the clarity and intuitiveness of the user interface and documentation.

### 3. Other comments

Fix: #3996

### CODE REVIEW

File: docs/man/events.1 [hunk 1]
```diff
@@ -49,8 +49,8 @@ Event return value: Filter events based on their return value using
 The event return value expression follows the syntax of a numerical
 expression.
 .IP \[bu] 2
-Event context fields: Filter events based on the non-argument fields
-defined in the trace.Event struct using `event-name.context.field'.
+Event scope fields: Filter events based on the non-argument fields
+defined in the trace.Event struct using `event-name.scope.field'.
```
Comment (geyslan) [inline docs/man/events.1:53]: I think that this assertion can be a bit misleading, I mean many of the former context fields are part of the trace.Event. WDYT?

File: pkg/cmd/flags/policy.go [hunk 2]
```diff
@@ -81,13 +81,13 @@ func PrepareFilterMapsFromPolicies(policies []k8s.PolicyInterface) (PolicyScopeM
 					continue
 				}
 
-				// at this point we know the filter is an event context filter
-				// context filters are provided without "context." prefix so we need to add it
-				evtContextFlags, err := parseEventFlag(fmt.Sprintf("%s.context.%s", r.Event, f))
+				// at this point we know the filter is an event scope filter
+				// scope filters are provided without "scope." prefix so we need to add it
+				evtScopeFlags, err := parseEventFlag(fmt.Sprintf("%s.scope.%s", r.Event, f))
```
Comment (geyslan) [inline pkg/cmd/flags/policy.go:86]: I think we should name it `evtScopeOpts` differentiating it from the `--scope` flags naming.

File: pkg/ebpf/events_pipeline.go [hunk 3]
```diff
@@ -325,8 +325,8 @@ func (t *Tracee) matchPolicies(event *trace.Event) uint64 {
 		// Do the userland filtering
 		//
 
-		// 1. event context filters
-		if !p.ContextFilter.Filter(*event) {
+		// 1. event scope filters
+		if !p.ScopeFilter.Filter(*event) {
```
Comment (geyslan) [inline pkg/ebpf/events_pipeline.go:329]:
```suggestion
		if !p.EventScopeFilter.Filter(*event) {
```
Based on my assumption of differentiating it above. WDYT?

**Review by geyslan — APPROVED** [review]
LGTM. Put just some naming suggestions.

File: docs/man/events.1 (hunk 1 above)
Comment (yanivagman) [reply docs/man/events.1:53]: We are renaming context filters to scope filters, and I didn't change anything in this comment.
Isn't it still true that scope filters are to filter non-argument fields in the trace.Event? Why is it misleading?

File: pkg/cmd/flags/policy.go (hunk 2 above)
Comment (yanivagman) [reply pkg/cmd/flags/policy.go:86]: I think having evt (enent) in the name already differentiates it from the "global" policy scope, does't it?

Comment (geyslan) [reply pkg/cmd/flags/policy.go:86]: You're right. I concur.

File: docs/man/events.1 (hunk 1 above)
Comment (geyslan) [reply docs/man/events.1:53]: Yep, it's not your changes, I just glanced at that sentence and, thinking as a new user, didn't quickly understand the meaning. We can see how to improve it in the future. It's definitely not related to this PR.

**Pull request merged**